cli/src/pcluster/models/cluster_config.py

```diff
@@ -765,8 +766,45 @@ def _validate(self):
     @property
     def architecture(self):
         """Compute cluster's architecture based on its head node instance type."""
-        instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
-        return instance_type_info.supported_architecture()[0]
+        return self._instance_type_info.supported_architecture()[0]
```

Comment:
we use functions in utils including some boto3 call, I think we should refactor them later to align with the AWSApi.

Reply:
Yes is in the TODO list, I'm currently focused on CDK migration to enable end-to-end tests, then I'll go back to fix/align all these remaining things. Thanks for pointing that out.